src/Helpers/Tokens/AsymmetricVerifier.php

```diff
@@ -0,0 +1,55 @@
+<?php
+declare(strict_types=1);
+
+namespace Auth0\SDK\Helpers\Tokens;
+
+use Auth0\SDK\Exception\InvalidTokenException;
+use Lcobucci\JWT\Signer\Key;
+use Lcobucci\JWT\Signer\Rsa\Sha256 as RsSigner;
+use Lcobucci\JWT\Token;
+
+/**
+ * Class AsymmetricVerifier
+ *
+ * @package Auth0\SDK\Helpers
+ */
+final class AsymmetricVerifier extends SignatureVerifier
+{
+
+    /**
+     * JWKS array with kid as keys, PEM cert as values.
+     *
+     * @var array
+     */
+    private $jwks;
+
+    /**
+     * JwksVerifier constructor.
+     *
+     * @param array $jwks JWKS to use.
+     */
+    public function __construct(array $jwks)
+    {
+        $this->jwks = $jwks;
+        parent::__construct('RS256');
+    }
+
+    /**
+     * Check the token kid and signature.
+     *
+     * @param Token $token Parsed token to check.
+     *
+     * @return boolean
+     *
+     * @throws InvalidTokenException If ID token kid was not found in the JWKS.
+     */
+    protected function checkSignature(Token $token) : bool
+    {
+        $tokenKid = $token->getHeader('kid', false);
+        if (! array_key_exists($tokenKid, $this->jwks)) {
```

Comment:
Would `! isset()` or `empty()` make more sense here? The current code is not type safe, given that the `Key` constructor requires a `string $content`:

https://github.com/lcobucci/jwt/blob/bdaf6ae2a78ded9b0a9d2c087f63948bc26d58cd/src/Signer/Key.php#L26

Reply:
Good catch but we're using the stable 3.3 branch and that's not a concern:

https://github.com/lcobucci/jwt/blob/3.3/src/Signer/Key.php#L34